src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts

```diff
@@ -104,25 +114,29 @@ export module DotNet {
    * Invokes the specified .NET public method synchronously. Not all hosting scenarios support
    * synchronous invocation, so if possible use invokeMethodAsync instead.
    *
+   * @deprecated Use DotNetObject to invoke instance methods instead.
    * @param assemblyName The short name (without key/version or .dll extension) of the .NET assembly containing the method.
    * @param methodIdentifier The identifier of the method to invoke. The method must have a [JSInvokable] attribute specifying this identifier.
    * @param args Arguments to pass to the method, each of which must be JSON-serializable.
    * @returns The result of the operation.
    */
   export function invokeMethod<T>(assemblyName: string, methodIdentifier: string, ...args: any[]): T {
-      return invokePossibleInstanceMethod<T>(assemblyName, methodIdentifier, null, args);
+      const dispatcher = getDefaultCallDispatcher();
+      return dispatcher.invokeDotNetStaticMethod<T>(assemblyName, methodIdentifier, ...args);
```

Comment:
We could make it so that calling this method throws an error unless explicitly allowed via some sort of option. That would encourage people to use `DotNetObjectReference` instead.